### DISCUSSION TITLE
Geocoding error

### DISCUSSION THREAD

**Discussion by exploreru** [gmtsar/gmtsar#897] [Q&A]
Hello everyone,I have a question that I would like to ask for you.when I am using ''p2p_S1_TOPS_Frame.csh'' to processing two images.I have found some unresolved issues.After I made modifications on region cut,I found that the ''los.grd'' file was not generated.

Linking files for Subswath 1 ...
Linking files for Subswath 2 ...
Linking files for Subswath 3 ...
[1] 65716
[2] 65717
[3] 65718
[3]  - Done                   p2p_processing.csh S1_TOPS  ...
[2]  - Done                   p2p_processing.csh S1_TOPS  ...
[1]  + Done                   p2p_processing.csh S1_TOPS  ...

Merging START
Number of Files to be merged is 3 
Number of Files to be merged is 3 
Number of Files to be merged is 3 
Merging END

MAKE LANDMASK -- START
REQUIRE FULL RESOLUTION COASTLINE FROM GMT

surface [WARNING]: 53478837 unusable points were supplied; these will be ignored.
surface [WARNING]: You should have pre-processed the data with block-mean, -median, or -mode.
surface [WARNING]: Check that previous processing steps write results with enough decimals.
surface [WARNING]: Possibly some data were half-way between nodes and subject to IEEE 754 rounding.
surface [WARNING]: 53478837 unusable points were supplied; these will be ignored.
surface [WARNING]: You should have pre-processed the data with block-mean, -median, or -mode.
surface [WARNING]: Check that previous processing steps write results with enough decimals.
surface [WARNING]: Possibly some data were half-way between nodes and subject to IEEE 754 rounding.
Killed
gmtinfo [ERROR]: No input data found!
xyz2grd [ERROR]: Syntax error: Must specify -R option
grdsample [ERROR]: Error for input file: No such file (landmask_ra.grd)
mv: Unable to obtain file status(stat)for''tmp.grd'':No file or directory availble
grd2xyz [ERROR]: File landmask_ra.grd not found
[Session gmt (0)]: Error returned from GMT API: GMT_FILE_NOT_FOUND (16)
grdinfo [ERROR]: Error for input file: No such file (landmask_ra.grd)
grdinfo [ERROR]: Syntax error: Must specify one or more input files
xyz2grd [ERROR]: Syntax error -I option: Must specify positive increment(s)
mv:  Unable to obtain file status(stat)for''tmp.grd'':No file or directory availble
MAKE LANDMASK -- END

SNAPHU.CSH - START
threshold_snaphu: 0.1
unwrapping phase with snaphu - higher threshold for faster unwrapping 

snaphu v2.0.4
32 parameters input from file /usr/local/GMTSAR/share/gmtsar/snaphu/config/snaphu.conf.brief (297 lines total)
Reading wrapped phase from file phase.in
No weight file specified.  Assuming uniform weights
Reading correlation data from file corr.in
Calculating smooth-solution cost parameters
Building range cost arrays
Building azimuth cost arrays
Initializing flows with MST algorithm
Running approximate minimum spanning tree solver
Running nonlinear network flow optimizer
Maximum flow on network: 5
Number of nodes in network: 4995502
Flow increment: 1  (Total improvements: 0)
Found 1 valid set(s) of connected pixels
Source 0 row, col = -2, -2
Treesize: 4995502    Pivots: 490418      Improvements: 3311       
Growing connected component mask
1 connected components formed
Writing connected components to file conncomp.out as 1-byte unsigned ints
Maximum flow on network: 1
Total solution cost: 8382608
Integrating phase
Writing output to file unwrap.out
Program snaphu done
Elapsed processor time:   0:00:11.60
Elapsed wall clock time:  0:00:12
Unwrapped phase map: unwrap.pdf
SNAPHU.CSH - END

GEOCODE-START
proj_ra2ll.csh
Sampling in geocoordinates with 50 meter pixels ...
proj_ra2ll.csh
Sampling in geocoordinates with 50 meter pixels ...
proj_ra2ll.csh
Sampling in geocoordinates with 50 meter pixels ...
Make phasefilt_ll.kml and phasefilt_ll.png
gs -q -dNOPAUSE -dBATCH -dNOSAFER -dSCANCONVERTERTYPE=2 -dMaxBitmap=2147483647 -dUseFastColor=true -dGraphicsAlphaBits=2 -dTextAlphaBits=4 -sDEVICE=pngalpha  -g5521x3518 -r1800 -sOutputFile='phasefilt_ll.png' './psconvert_68448d.eps'
Make phasefilt_mask_ll.kml and phasefilt_mask_ll.png
gs -q -dNOPAUSE -dBATCH -dNOSAFER -dSCANCONVERTERTYPE=2 -dMaxBitmap=2147483647 -dUseFastColor=true -dGraphicsAlphaBits=2 -dTextAlphaBits=4 -sDEVICE=pngalpha  -g5521x3518 -r1800 -sOutputFile='phasefilt_mask_ll.png' './psconvert_68464d.eps'
grdimage [WARNING]: Your image is grayscale only but -Q requires 24-bit; image will be expanded to 24-bit.
PSL: Colormap of 255 colors created
Make corr_ll.kml and corr_ll.png
gs -q -dNOPAUSE -dBATCH -dNOSAFER -dSCANCONVERTERTYPE=2 -dMaxBitmap=2147483647 -dUseFastColor=true -dGraphicsAlphaBits=2 -dTextAlphaBits=4 -sDEVICE=pngalpha  -g5521x3518 -r1800 -sOutputFile='corr_ll.png' './psconvert_68479d.eps'
grdmath [ERROR]: grid files not of same size!
grdmath [ERROR]: Syntax error: unwrap_mask.grd is not a number, operator or file name
proj_ra2ll.csh
Sampling in geocoordinates with 50 meter pixels ...
proj_ra2ll.csh
grd2xyz [ERROR]: File unwrap_mask.grd not found
[Session gmt (0)]: Error returned from GMT API: GMT_FILE_NOT_FOUND (16)
Sampling in geocoordinates with 50 meter pixels ...
gmtinfo [ERROR]: No input data found!
range: Subscript out of range.
gmtmath [ERROR]: Syntax error: Operation "DIV" requires 2 operands
gmtmath [ERROR]: Syntax error: Operation "MUL" requires 2 operands
gmtinfo [ERROR]: No input data found!
blockmedian [ERROR]: Unable to decode  as a floating point number
xyz2grd [ERROR]: Unable to decode  as a floating point number
proj_ra2ll.csh
grd2xyz [ERROR]: File los.grd not found
[Session gmt (0)]: Error returned from GMT API: GMT_FILE_NOT_FOUND (16)
Sampling in geocoordinates with 50 meter pixels ...
gmtinfo [ERROR]: No input data found!
range: Subscript out of range.
gmtmath [ERROR]: Syntax error: Operation "DIV" requires 2 operands
gmtmath [ERROR]: Syntax error: Operation "MUL" requires 2 operands
gmtinfo [ERROR]: No input data found!
blockmedian [ERROR]: Unable to decode  as a floating point number
xyz2grd [ERROR]: Unable to decode  as a floating point number
grdinfo [ERROR]: Error for input file: No such file (unwrap_mask_ll.grd)
grdinfo [ERROR]: Syntax error: Must specify one or more input files
gmtmath [ERROR]: Syntax error: Operation "INV" requires 1 operands
grdimage [ERROR]: Error for input file: No such file (unwrap_mask_ll.grd)
grdimage [ERROR]: Syntax error: Must specify one (or three) input file(s)
Make unwrap_mask_ll.kml and unwrap_mask_ll.png
psconvert [ERROR]: The file unwrap_mask_ll.ps has no BoundingBox in the first 20 lines or last 256 bytes. Use -A option.
Make unwrap_ll.kml and unwrap_ll.png
gs -q -dNOPAUSE -dBATCH -dNOSAFER -dSCANCONVERTERTYPE=2 -dMaxBitmap=2147483647 -dUseFastColor=true -dGraphicsAlphaBits=2 -dTextAlphaBits=4 -sDEVICE=pngalpha  -g1710x1118 -r1800 -sOutputFile='unwrap_ll.png' './psconvert_68571d.eps'
grdinfo [ERROR]: Error for input file: No such file (los.grd)
grdinfo [ERROR]: Syntax error: Must specify one or more input files
grdinfo [ERROR]: Error for input file: No such file (los.grd)
grdinfo [ERROR]: Syntax error: Must specify one or more input files
makecpt [ERROR]: Syntax error -T: Must specify valid min[/max[/inc[<unit>|+n]]] option
grdinfo [ERROR]: Error for input file: No such file (los_ll.grd)
grdinfo [ERROR]: Syntax error: Must specify one or more input files
gmtmath [ERROR]: Syntax error: Operation "INV" requires 1 operands
grdimage [ERROR]: Error for input file: No such file (los_ll.grd)
grdimage [ERROR]: Syntax error: Must specify one (or three) input file(s)
Make los_ll.kml and los_ll.png
psconvert [ERROR]: The file los_ll.ps has no BoundingBox in the first 20 lines or last 256 bytes. Use -A option.
GEOCODE END
rm: No match.

**Comment by Xiaohua-Eric-Xu** [comment]
`Killed ` This might indicate a lack of memory, but strangely you were able to process three subswaths at the same time but are not able to create landmask which theoretically costs much less memory.

**Reply by exploreru** [reply]
Thanks.Your suggestion is appropriate and beneficial.After I increase the memory,it worked successfully.
but It prompts as follows.

MAKE LANDMASK -- START
REQUIRE FULL RESOLUTION COASTLINE FROM GMT

surface [WARNING]: 53478837 unusable points were supplied; these will be ignored.
surface [WARNING]: You should have pre-processed the data with block-mean, -median, or -mode.
surface [WARNING]: Check that previous processing steps write results with enough decimals.
surface [WARNING]: Possibly some data were half-way between nodes and subject to IEEE 754 rounding.
surface [WARNING]: 53478837 unusable points were supplied; these will be ignored.
surface [WARNING]: You should have pre-processed the data with block-mean, -median, or -mode.
surface [WARNING]: Check that previous processing steps write results with enough decimals.
surface [WARNING]: Possibly some data were half-way between nodes and subject to IEEE 754 rounding.
MAKE LANDMASK -- END

I guess that it may be caused by the wrong 'region_cut' parameter value.

**Reply by Xiaohua-Eric-Xu** [reply]
These are GMT Warnings, which can usually be neglected. What's listed above means you have some data outside region or maybe multiple data inside one cell.

**Reply by exploreru** [reply]
Okay，I understand now.but Something goes wrong in the 'GEOCODE' section.It prompts me as follows.

GEOCODE-START
proj_ra2ll.csh
surface [WARNING]: Your grid dimensions are mutually prime.  Convergence is very unlikely.
surface [WARNING]: 57605313 unusable points were supplied; these will be ignored.
surface [WARNING]: You should have pre-processed the data with block-mean, -median, or -mode.
surface [WARNING]: Check that previous processing steps write results with enough decimals.
surface [WARNING]: Possibly some data were half-way between nodes and subject to IEEE 754 rounding.
surface [WARNING]: Your grid dimensions are mutually prime.  Convergence is very unlikely.
surface [WARNING]: 57605313 unusable points were supplied; these will be ignored.
surface [WARNING]: You should have pre-processed the data with block-mean, -median, or -mode.
surface [WARNING]: Check that previous processing steps write results with enough decimals.
surface [WARNING]: Possibly some data were half-way between nodes and subject to IEEE 754 rounding.
Sampling in geocoordinates with 50 meter pixels ...
proj_ra2ll.csh
Sampling in geocoordinates with 50 meter pixels ...
proj_ra2ll.csh
Sampling in geocoordinates with 50 meter pixels ...
Make phasefilt_ll.kml and phasefilt_ll.png
gs -q -dNOPAUSE -dBATCH -dNOSAFER -dSCANCONVERTERTYPE=2 -dMaxBitmap=2147483647 -dUseFastColor=true -dGraphicsAlphaBits=2 -dTextAlphaBits=4 -sDEVICE=pngalpha  -g5521x3518 -r1800 -sOutputFile='phasefilt_ll.png' './psconvert_10127d.eps'
Make phasefilt_mask_ll.kml and phasefilt_mask_ll.png
gs -q -dNOPAUSE -dBATCH -dNOSAFER -dSCANCONVERTERTYPE=2 -dMaxBitmap=2147483647 -dUseFastColor=true -dGraphicsAlphaBits=2 -dTextAlphaBits=4 -sDEVICE=pngalpha  -g5521x3518 -r1800 -sOutputFile='phasefilt_mask_ll.png' './psconvert_10142d.eps'
grdimage [WARNING]: Your image is grayscale only but -Q requires 24-bit; image will be expanded to 24-bit.
PSL: Colormap of 255 colors created
Make corr_ll.kml and corr_ll.png
gs -q -dNOPAUSE -dBATCH -dNOSAFER -dSCANCONVERTERTYPE=2 -dMaxBitmap=2147483647 -dUseFastColor=true -dGraphicsAlphaBits=2 -dTextAlphaBits=4 -sDEVICE=pngalpha  -g5521x3518 -r1800 -sOutputFile='corr_ll.png' './psconvert_10157d.eps'
grdmath [ERROR]: grid files not of same size!
grdmath [ERROR]: Syntax error: unwrap_mask.grd is not a number, operator or file name
proj_ra2ll.csh
Sampling in geocoordinates with 50 meter pixels ...
proj_ra2ll.csh
grd2xyz [ERROR]: File unwrap_mask.grd not found
[Session gmt (0)]: Error returned from GMT API: GMT_FILE_NOT_FOUND (16)
Sampling in geocoordinates with 50 meter pixels ...
gmtinfo [ERROR]: No input data found!
range: Subscript out of range.
gmtmath [ERROR]: Syntax error: Operation "DIV" requires 2 operands
gmtmath [ERROR]: Syntax error: Operation "MUL" requires 2 operands
gmtinfo [ERROR]: No input data found!
blockmedian [ERROR]: Unable to decode  as a floating point number
xyz2grd [ERROR]: Unable to decode  as a floating point number
proj_ra2ll.csh
grd2xyz [ERROR]: File los.grd not found
[Session gmt (0)]: Error returned from GMT API: GMT_FILE_NOT_FOUND (16)
Sampling in geocoordinates with 50 meter pixels ...
gmtinfo [ERROR]: No input data found!
range: Subscript out of range.
gmtmath [ERROR]: Syntax error: Operation "DIV" requires 2 operands
gmtmath [ERROR]: Syntax error: Operation "MUL" requires 2 operands
gmtinfo [ERROR]: No input data found!
blockmedian [ERROR]: Unable to decode  as a floating point number
xyz2grd [ERROR]: Unable to decode  as a floating point number
grdinfo [ERROR]: Error for input file: No such file (unwrap_mask_ll.grd)
grdinfo [ERROR]: Syntax error: Must specify one or more input files
gmtmath [ERROR]: Syntax error: Operation "INV" requires 1 operands
grdimage [ERROR]: Error for input file: No such file (unwrap_mask_ll.grd)
grdimage [ERROR]: Syntax error: Must specify one (or three) input file(s)
Make unwrap_mask_ll.kml and unwrap_mask_ll.png
psconvert [ERROR]: The file unwrap_mask_ll.ps has no BoundingBox in the first 20 lines or last 256 bytes. Use -A option.
Make unwrap_ll.kml and unwrap_ll.png
gs -q -dNOPAUSE -dBATCH -dNOSAFER -dSCANCONVERTERTYPE=2 -dMaxBitmap=2147483647 -dUseFastColor=true -dGraphicsAlphaBits=2 -dTextAlphaBits=4 -sDEVICE=pngalpha  -g1710x1118 -r1800 -sOutputFile='unwrap_ll.png' './psconvert_10249d.eps'
grdinfo [ERROR]: Error for input file: No such file (los.grd)
grdinfo [ERROR]: Syntax error: Must specify one or more input files
grdinfo [ERROR]: Error for input file: No such file (los.grd)
grdinfo [ERROR]: Syntax error: Must specify one or more input files
makecpt [ERROR]: Syntax error -T: Must specify valid min[/max[/inc[<unit>|+n]]] option
grdinfo [ERROR]: Error for input file: No such file (los_ll.grd)
grdinfo [ERROR]: Syntax error: Must specify one or more input files
gmtmath [ERROR]: Syntax error: Operation "INV" requires 1 operands
grdimage [ERROR]: Error for input file: No such file (los_ll.grd)
grdimage [ERROR]: Syntax error: Must specify one (or three) input file(s)
Make los_ll.kml and los_ll.png
psconvert [ERROR]: The file los_ll.ps has no BoundingBox in the first 20 lines or last 256 bytes. Use -A option.
GEOCODE END
rm: No match.

It seems that File ''unwrap_mask.grd'' was not generated.
Is it because I made modifications on region_cut parameter?
When I leaved it blank the programme worked normally.

**Reply by Xiaohua-Eric-Xu** [reply]
It could be. The current region_cut in the config file will cut the SLC directly. If you run snaphu yourself, there's another region_cut you can append to make the region smaller. If you ran that by yourself but corr.grd remains the original size, then there could be a problem later on as the size of your corr.grd dose not match unwrap.grd. You may replace your corr.grd with the following commands:
```
mv corr.grd corr_save.grd
gmt grdcut corr_save.grd -Runwrap.grd -Gcorr.grd
```
